src/viewer/scene/CameraControl/CameraControl.js

```diff
@@ -1444,6 +1444,20 @@ class CameraControl extends Component {
      */
     set panRightClick(value) {
         this._configs.panRightClick = value !== false;
+        const panKeyMap = this._keyMap[this.MOUSE_PAN];
+        if (panKeyMap && panKeyMap.length > 0) {
+            const input = this.scene.input;
+            for (let i = 0, len = panKeyMap.length; i < len; i++) {
+                if (this._configs.panRightClick && panKeyMap[i] === input.MOUSE_MIDDLE_BUTTON) {
+                    this._keyMap[this.MOUSE_PAN][i] = input.MOUSE_RIGHT_BUTTON;
+                    return;
```

Comment:
Let's remove these return statements, to make robust for the case when panKeyMap contains duplicate values, so all of those can get replaced